modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/UserCardTag.java

```diff
@@ -0,0 +1,584 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.frontend.taglib.clay.servlet.taglib;
+
+import com.liferay.frontend.taglib.clay.internal.servlet.taglib.BaseContainerTag;
+import com.liferay.frontend.taglib.clay.servlet.taglib.soy.UserCard;
+import com.liferay.frontend.taglib.clay.servlet.taglib.util.DropdownItem;
+import com.liferay.frontend.taglib.clay.servlet.taglib.util.LabelItem;
+import com.liferay.petra.string.StringBundler;
+import com.liferay.portal.kernel.theme.ThemeDisplay;
+import com.liferay.portal.kernel.util.ListUtil;
+import com.liferay.portal.kernel.util.Validator;
+import com.liferay.portal.kernel.util.WebKeys;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import javax.servlet.jsp.JspException;
+import javax.servlet.jsp.JspWriter;
+
+/**
+ * @author Carlos Lancha
+ */
+public class UserCardTag extends BaseContainerTag {
+
+	@Override
+	public int doStartTag() throws JspException {
+		setAttributeNamespace(_ATTRIBUTE_NAMESPACE);
+
+		ThemeDisplay themeDisplay = (ThemeDisplay)request.getAttribute(
+			WebKeys.THEME_DISPLAY);
+
+		String pathThemeImages = themeDisplay.getPathThemeImages();
+
+		_spritemap = pathThemeImages.concat("/clay/icons.svg");
+
+		return super.doStartTag();
+	}
+
+	public List<DropdownItem> getActionDropdownItems() {
+		if ((_actionDropdownItems == null) && (_userCard != null)) {
+			return _userCard.getActionDropdownItems();
+		}
+
+		return _actionDropdownItems;
+	}
+
+	@Override
+	public String getCssClass() {
+		if ((super.getCssClass() == null) && (_userCard != null)) {
+			return _userCard.getCssClass();
+		}
+
```

Comment:
We need to add here check for `elementClasses`. This is not edge, but the most common case: we are passing `Card` object to tag, and in the `Card` object we have only `elementClasses` set.

I did this:
```
@Override
public String getCssClass() {
	if (super.getCssClass() == null &&
		Validator.isNotNull(_horizontalCard)) {

		if (_horizontalCard.getCssClass() != null) {
			return _horizontalCard.getCssClass();
		}

		if (_horizontalCard.getElementClasses() != null) {
			return _horizontalCard.getElementClasses();
		}
	}

	return super.getCssClass();
}
```
Alternative would we to handle this in `prepareProps` of the base class.

Reply:
We have to do it here, yes, otherwise it would only affect to react rendering and not server side. I'll check it, thx!